src/base.ts

```diff
@@ -78,6 +79,16 @@ export default abstract class extends Command {
     }
   }
 
+  async setSource() {
+    const specFilePath = this.specFile?.getFilePath();
+    if (!specFilePath) { return; }
+    try {
+      const stats = await stat(specFilePath);
+      this.metricsMetadata['file_creation_timestamp'] = stats.birthtimeMs;
+    } catch (e: any) {
+      // do nothing.
```

Comment:
What are the possible errors we could find here? File doesn't exist maybe? We don't have permission to access the file?

I have the feeling we should control (and report to New Relic) failures here.

Reply:
Since this function is ultimately being [called from](https://github.com/KhudaDad414/cli/blob/aca055c076769d36978a81bc5714adccfa2c6a36/src/base.ts#L95-L96) the `finally` method of the `command` with a file path, If there is any error with the file, We expect that the current call is actually reporting the error. 

clarified the comment in the code.

Reply:
Just one question on this topic: if there is an error with the file, is it handled at the `catch` [method](https://github.com/KhudaDad414/cli/blob/aca055c076769d36978a81bc5714adccfa2c6a36/src/base.ts#L30-L39) as well, apart from the `finally` method? 🤔

Reply:
@peter-rr if by "handled" you mean reporting to New Relic, then it is only being done in `finally` since it will run regardless of errors.

Reply:
@KhudaDad414 @peter-rr It's generally considered best practice to handle errors as close to their source as possible. This helps maintain clear code structure and makes it easier to understand and reason about error handling logic, in this case having some exception type checks under the `catch` would help.

Reply:
@Amzani this is how I understand the CLI would work when an error happens:

1) A CLI command is run against an invalid file path, or a file that the user doesn't have read permission etc...
2) Error is thrown.
2) `catch` method will log the error.
3) `finally` method will report it to New Relic.

`setSouce` is being called from the `finally` method. so the same error that caused the `finally` method to run will occur again. If the error is already logged and is being reported to New Relic, I am failing to understand why it should be handled and not be ignored here.

Reply:
I would need some clarification here: are we all talking about the `catch` [method](https://github.com/asyncapi/cli/pull/1304/files/aca055c076769d36978a81bc5714adccfa2c6a36#diff-440b55a9ff9e0486020fc49197f3278d14446306fc0c5d5107b48139c20bfda6R30-R39) from the class, or the `catch` [block](https://github.com/asyncapi/cli/pull/1304/files/aca055c076769d36978a81bc5714adccfa2c6a36#diff-440b55a9ff9e0486020fc49197f3278d14446306fc0c5d5107b48139c20bfda6R88-R91) inside the `setSource` method?

Reply:
I am saying that if there is an error with the file, we expect the command to throw an error, `catch` method inside the class will log the error, and `finally` method will report it.

Since The `setSource` is called from the `finally` method, it is unnecessary to do anything inside of it's `catch` block. 

If we log it, we have logged it twice, if we initiate a report we have reported it twice.

One time by command trying to access the file and throwing the error and one time by us trying to access it here.


Hope it is clear now.